app/src/main/java/pl/droidevs/books/library/LibraryActivity.java

```diff
@@ -1,15 +1,77 @@
 package pl.droidevs.books.library;
 
+import android.arch.lifecycle.ViewModelProvider;
+import android.arch.lifecycle.ViewModelProviders;
 import android.os.Bundle;
+import android.support.design.widget.FloatingActionButton;
 import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.DefaultItemAnimator;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.widget.ProgressBar;
 
+import javax.inject.Inject;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.Unbinder;
+import dagger.android.AndroidInjection;
 import pl.droidevs.books.R;
 
+import static android.view.View.GONE;
+import static android.view.View.VISIBLE;
+
 public class LibraryActivity extends AppCompatActivity {
 
+    @BindView(R.id.layout_books)
+    RecyclerView recyclerView;
+
+    @BindView(R.id.progress_books)
+    ProgressBar progressBar;
+
+    @BindView(R.id.button_add_book)
+    FloatingActionButton floatingActionButton;
+
+    @Inject
+    ViewModelProvider.Factory viewModelFactory;
+
+    private Unbinder unbinder;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_library);
+
+        AndroidInjection.inject(this);
+        this.unbinder = ButterKnife.bind(this);
+
+        final LibraryAdapter adapter = new LibraryAdapter();
+        adapter.setItemClickListener(bookId -> {
+            // TODO: Start details activity and pass the book id
+        });
+        recyclerView.setAdapter(adapter);
+        recyclerView.setHasFixedSize(true);
+        recyclerView.setLayoutManager(new LinearLayoutManager(this));
+        recyclerView.setItemAnimator(new DefaultItemAnimator());
+
+        floatingActionButton.setOnClickListener(view -> {
```

Comment:
Nie wiem, czy to jest potrzebne w onCrete - butterknife udostępnia przecież @OnClick

Reply:
Ja osobiście nie lubię tych metod @OnClick. Wydaje mi się, że było one użyteczne, gdy lambdy nie były jeszcze dostępne i należało tworzyć anonimowe klasy. Wydaje mi się, że znacznie bardziej czytelne, gdy przy deklaracji przycisku zdefiniujesz mu lambdę z listenerem niż stworzysz w klasie metodę, która obsługuje kliknięcia. Potem te metody się mnożą, na pierwszy rzut oka trudno określić, który przycisk obsługują.

Reply:
Nie upieram się - to kwestia preferencji. Ja nie lubię długich rozbudowanych metod, więc i tak bym tworzyła jakąś metodę, która "zajęłaby się" guzikiem. Dlatego też @OnClick tutaj dla mnie jest idealny :)